crates/sui-benchmark/src/benchmark_setup.rs

```diff
@@ -194,76 +193,84 @@ impl Env {
+            }
+            fullnodes
         } else {
             info!("Using LocalValidatorAggregatorProxy");
             let reconfig_fullnode_rpc_url =
                 if use_fullnode_for_reconfig {
-                    Some(fullnode_rpc_url.expect(
+                    // Only need to use one full node for reconfiguration.
+                    Some(fullnode_rpc_urls.get(0).expect(
                         "fullnode-rpc-url is required when use-fullnode-for-reconfig is true",
                     ))
                 } else {
                     None
                 };
             let genesis = sui_config::node::Genesis::new_from_file(genesis_blob_path);
             let genesis = genesis.genesis()?;
-            Arc::new(
+            vec![Arc::new(
                 LocalValidatorAggregatorProxy::from_genesis(
                     genesis,
                     registry,
-                    reconfig_fullnode_rpc_url.as_deref(),
+                    reconfig_fullnode_rpc_url.map(|x| &**x),
                 )
                 .await,
-            )
+            )]
         };
         info!(
             "Reconfiguration - Reconfiguration to epoch {} is done",
-            proxy.get_current_epoch(),
+            proxies[0].get_current_epoch(),
         );
 
-        let offset = ObjectID::from_hex_literal(primary_gas_id)?;
-        let ids = ObjectID::in_range(offset, primary_gas_objects)?;
-        let primary_gas_id = ids
-            .choose(&mut rand::thread_rng())
-            .context("Failed to choose a random primary gas id")?;
-        let primary_gas = proxy.get_object(*primary_gas_id).await?;
-        let pay_coin_id = ids
-            .choose(&mut rand::thread_rng())
-            .context("Failed to choose a random pay coin")?;
-        let pay_coin = proxy.get_object(*pay_coin_id).await?;
-        let primary_gas_account = primary_gas.owner.get_owner_address()?;
-        let keystore_path = Some(&keystore_path)
-            .filter(|s| !s.is_empty())
-            .map(PathBuf::from)
-            .ok_or_else(|| {
-                anyhow!(format!(
-                    "Failed to find keypair at path: {}",
-                    &keystore_path
-                ))
-            })?;
-        let keypair = Arc::new(get_ed25519_keypair_from_keystore(
-            keystore_path,
-            &primary_gas_account,
-        )?);
-        let ttag = pay_coin.get_move_template_type()?;
-        Ok(BenchmarkSetup {
-            primary_gas: (
-                primary_gas.compute_object_reference(),
-                Owner::AddressOwner(primary_gas_account),
-                keypair.clone(),
-            ),
-            pay_coin: (pay_coin.compute_object_reference(), pay_coin.owner, keypair),
-            pay_coin_type_tag: ttag,
-            validator_proxy: proxy,
-        })
+        let mut proxy_gas_and_coins = vec![];
+
+        for proxy in proxies.iter() {
+            let offset = ObjectID::from_hex_literal(primary_gas_id)?;
```

Comment:
It's unclear to me why we need to go through all proxies to generate gas. Any one should be enough, no?

Reply:
After this week, looks like scaling this is not needed anymore urgently. But the idea was that if we use a separate gas coin for each proxy we can then both generate gas coins in parallel for each proxy  without contention over primary gas id (if its a bottleneck) and we can ensure that we don't have any issues splitting gas coins if the load + num of proxies is larger than the total amount of one shared primary gas/pay coin.